Source/Project/Sizer/Sizer.f90

```diff
@@ -319,8 +341,24 @@ FUNCTION sizeFunctionMinimumOnBox( sizer, xMin, xMax ) RESULT(hMin)
                DO i = 0, nX(1)
                   x(1) = xMin(1) + i*dx(1)
                   hMin = MIN( hMin, controlSize(sizer,x) )
-               END DO
+              END DO
             END DO
+         END IF
+!
+!        -----------------------------------------
+!        Sizes determined by the bottom topography
+!        -----------------------------------------
+!
+         IF ( ASSOCIATED(sizer % topography) )     THEN
+            dX = (xMax - xMin)/nX
+            TLoop : DO j = 0, nX(2)
+               x(2) = xMin(2) + j*dx(2)
+               DO i = 0, nX(1)
+                  x(1) = xMin(1) + i*dx(1)
+                  hMin = MIN( hMin, 1.0_RP/SQRT(sizer % topography % gaussianCurvatureAt(x)) )
```

Comment:
`gaussianCurvatureAt` is a procedure pointer to `gaussianCurvatureBaseAt`, is the plan to have this use the `GaussianCurvature` interface instead? Right now none of the routines in `GaussianCurvature.f90` are called in the testing.

Reply:
The expectation is that this might/should be overloaded in subclasses where perhaps the FD approximation of the hessian wouldn't be needed. The base class should (also) access the GC module, see the first comment above. (As an aside, I've also thought about moving your bi-cubic interpolation into its own class, and then compute an interplant for the GC, which would be used in the sizer. I'd think this might be faster than computing the hessian at every point every time it is needed.)

Reply:
Moving the bi-cubic spline could be a good idea. In essence it is a general approximation "tool" rather than a specific bottom topography algorithm. Separating it might also make is easier to generalize and test the functions for data that is not gridded.

Reply:
Although if we plan to do this it should live in a different pull request

Reply:
Yes. Let me look at it this weekend. In a new branch, I'll move the interpolation into its own class, and then in the "bottomtopographyfromfile" I'll add the interplolant of the gaussian curvature, and its own function for computing the curvature at (x,y) using the interplant. It will be interesting to compare speed.